src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -32,71 +34,84 @@ export default class GooglePayPaymentStrategy implements PaymentStrategy {
         private _googlePayAdyenV2PaymentProcessor?: GooglePayAdyenV2PaymentProcessor
     ) {}
 
-    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
-        this._methodId = options.methodId;
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options;
 
-        return this._googlePayPaymentProcessor.initialize(this._methodId)
-            .then(() => {
-                this._googlePayOptions = this._getGooglePayOptions(options);
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId));
+        this._paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
 
-                if (!this._googlePayOptions) {
-                    throw new InvalidArgumentError('Unable to initialize payment because "options.googlepay" argument is not provided.');
-                }
+        this._googlePayOptions = this._getGooglePayOptions(options);
 
-                const walletButton = this._googlePayOptions.walletButton && document.getElementById(this._googlePayOptions.walletButton);
+        this._buttonClickEventHandler = this._handleButtlonClickedEvent(methodId);
 
-                if (walletButton) {
-                    this._walletButton = walletButton;
-                    this._walletButton.addEventListener('click', this._handleWalletButtonClick);
-                }
+        if (this._paymentMethod.initializationData.nonce) {
+            return Promise.resolve(this._store.getState());
+        }
 
-                return this._store.getState();
-            });
+        await this._googlePayPaymentProcessor.initialize(methodId);
+
+        if (!this._googlePayOptions.walletButton) {
+            throw new InvalidArgumentError('walletButton argument is missing');
+        }
+
+        const walletButton = document.getElementById(this._googlePayOptions.walletButton);
+
+        if (!walletButton) {
+            throw new InvalidArgumentError('Unable to create wallet, walletButton ID could not be found');
+        }
+
+        this._walletButton = walletButton;
+        this._walletButton.addEventListener('click', this._buttonClickEventHandler);
+
+        return Promise.resolve(this._store.getState());
```

Comment:
whenever you are working with `async` methods, you don't need to wrap the return value in `Promise.resolve()`.

Async methods automatically wrap returned values in a promise, and thrown errors are wrapped in a `Promise.reject()`